src/lib/sw_engine/tvgSwRaster.cpp

```diff
@@ -74,6 +74,24 @@ struct FillRadial
     }
 };
 
+struct FillRadialEdgeCase
```

Comment:
@hermet I added this, because I wanted to check asap, which 'fillRadial' function to call. I didn't want to add any checks inside the fillRadial and I needed to distinguish between cases a == 0 and a > 0.

Reply:
I didn't expect the methods to be separated (RadialFill  vs RadialFillEdgeCase). I suppose the focal version could encompass the previous radial gradient behavior mathematically.

Reply:
@hermet please check the second commit - now the check ```if (fill->radial.a < FLT_EPSILON)``` is done every time fillRadial is called. my plan was to avoid it. 
do you prefer the solution from the second commit? (I'll squash them)